commit 25c9a0e0210232c8cf7ad8a60cd4cfd07db58879 (refs/changes/94/2894/1)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2017-11-01T15:42:36+00:00 (1 year, 11 months ago)
    
    OS-6433 zfs should panic if dp_dirty_total would go negative

diff --git a/usr/src/uts/common/fs/zfs/dsl_pool.c b/usr/src/uts/common/fs/zfs/dsl_pool.c
index 9c4fb790f2..266359aad9 100644
--- a/usr/src/uts/common/fs/zfs/dsl_pool.c
+++ b/usr/src/uts/common/fs/zfs/dsl_pool.c
@@ -724,7 +724,7 @@ dsl_pool_undirty_space(dsl_pool_t *dp, int64_t space, uint64_t txg)
 	}
 	ASSERT3U(dp->dp_dirty_pertxg[txg & TXG_MASK], >=, space);
 	dp->dp_dirty_pertxg[txg & TXG_MASK] -= space;
-	ASSERT3U(dp->dp_dirty_total, >=, space);
+	VERIFY3U(dp->dp_dirty_total, >=, space);
 	dsl_pool_dirty_delta(dp, -space);
 	mutex_exit(&dp->dp_lock);
 }
